compiler/rustc_resolve/src/late.rs

```diff
@@ -2417,6 +2417,9 @@ impl<'a, 'ast, 'ra, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
                         trace!("FindReferenceVisitor inserting res={:?}", lt_res);
                         self.lifetime.insert(lt_res);
                     }
+                } else if let TyKind::ImplTrait(..) = ty.kind {
+                    // Do not attempt to look inside impl-trait.
```

Comment:
Could you make these FIXMEs that briefly mention the "more principled fix" you mention in the PR description?